### ISSUE TITLE
Install timescaledb-toolkit on Windows

### ISSUE THREAD

**Issue by Cdkhanna** [timescale/timescaledb-toolkit#588]
**Relevant system information:**
 - OS:  Windows 10 x64
 - PostgreSQL version (output of `SELECT version();`): 13.2
 - TimescaleDB Toolkit version (output of `\dx timescaledb_toolkit` in `psql`): 2.8.1

**Describe the bug**
Unable to install timescaledb-toolkit on windows sever. There is not method I could find to install it on Windows.

Thanks

**Comment by davidkohn88** [comment]
Hi! Yes, we currently don't support Windows. I've changed this to an enhancement label, and people can upvote here and we can see how to prioritize based on that. Thanks for filing this!

**Comment by davetapley** [comment]
@davidkohn88 could current lack of Windows support for toolkit be made more explicit over here?
- https://github.com/timescale/timescaledb#installation
- https://docs.timescale.com/api/latest/hyperfunctions/

**Comment by davidkohn88** [comment]
@charislam do you think you could make the updates?

Also, we'd probably recommend as a workaround running PG in a Docker container on Windows. Just as a note.

**Comment by charislam** [comment]
> @charislam do you think you could make the updates?

Done: https://github.com/timescale/docs/pull/1837

**Comment by agarciagan** [comment]
Hi,

Is there any news about installing Toolkit on Windows? Windows support for Toolkit is under expected functionalitites of next releases, or is it something not expected within 2023?

Thanks

**Comment by jerryxwu** [comment]
@iroussos that's correct. We have upstream dependency that we cannot workaround around ATM.

**Comment by ConveyOS** [comment]
Any updates?

**Comment by sherlock1982** [comment]
@jerryxwu can you please elaborate what are the issues for building on Windows? 
Can we have at least a limited functionality with some functions throwing exceptions?

**Comment by bcemmett** [comment]
Was the issue with Windows builds mainly the dependency on PGX mentioned [here](https://github.com/timescale/timescaledb-toolkit/discussions/447#discussioncomment-2949945)? Asking because it looks like [as of last month that does now support Windows](https://github.com/pgcentralfoundation/pgrx/pull/2040), so I wondered if that now makes a Windows build possible?

**Comment by GrandMarch** [comment]
Currently GPRX supports Windows. So recently I tried to compile this project, but it actually failed, with package dependency issues. My level is not very good, and I have not been able to solve it.

**Comment by GrandMarch** [comment]
i think this issuse could be closed.i had complied this repo successfuly on windows ,by using pgrx 0.16.1

**Comment by iroussos** [comment]
@bcemmett I am not affiliated with Timescale any more, I'll let someone with more knowledge of the roadmap respond.

FYI @philkra on the question above

**Comment by miinesct** [comment]
> i think this issuse could be closed.i had complied this repo successfuly on windows ,by using pgrx 0.16.1

@GrandMarch Great to hear you got it compiled on Windows with pgrx 0.16.1 – nice work!

Could you share a quick step-by-step of what dependencies you had to install first on Windows (like pgrx, Rust, Visual Studio, etc.)? That would really help others.

Happy to buy you a coffee (or pay you) for your trouble – just let me know how.

Using time_weight with LOCF makes average calculations way easier in industrial IoT, especially with missing data. So having good Windows support is super valuable.

Thanks again!

**Comment by juneleung** [comment]
> i think this issuse could be closed.i had complied this repo successfuly on windows ,by using pgrx 0.16.1

Could you share some details, Thank you soo much!

**Comment by wuyuanyi135** [comment]
@juneleung  i let opencode+deepseek agent to make the MSVC timescaledb_toolkit build and it successfully run with tsdb 18, see my fork https://github.com/wuyuanyi135/timescaledb-toolkit

**Comment by juneleung** [comment]
> [@juneleung](https://github.com/juneleung) i let opencode+deepseek agent to make the MSVC timescaledb_toolkit build and it successfully run with tsdb 18, see my fork https://github.com/wuyuanyi135/timescaledb-toolkit

Thank you!
